pkgs/games/starruler2/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  callPackage,
+  makeWrapper,
+  # Path to directory containing music for the game,
+  # such as `data/music` from the commercial version.
+  musicPath ? null,
+  # Paths of mods to add.
+  modPaths ? [],
+}: let
+  data = callPackage ./data.nix {};
+  engine = callPackage ./engine.nix {};
+in
+  # The game cannot access files outside the working directory,
+  # so it cannot access files behind symlinks.
+  stdenv.mkDerivation {
+    pname = "starruler2";
+    version = lib.trivial.max data.version engine.version;
+
+    phases = ["installPhase"];
+
+    nativeBuildInputs = [makeWrapper];
+
+    installPhase = ''
+      game="$out/share/games/starruler2"
+      mkdir -p "$out/bin"
+
+      cp -r "${data}/share" "$out/share"
+      chmod -R u+w "$out/share"
+
+      ${
```

Comment:
Why not just use `lib.optionalString` like I did ? It would make this much more readable

Reply:
Done.

Reply:
Nice. And is there any reason why you prefer to do this:
```
''
some stuff

${
  lib.optionalString cond str
}
''
```

instead of this ?
```
''
some stuff
'' + lib.optionalString cond str
```
(with wherever alejandra places the +)

I think the later is cleaner